web-components/src/components/form/BasketTakeForm.tsx

```diff
@@ -0,0 +1,224 @@
+
+import { Theme } from '../../theme/muiTheme';
+import AmountField from '../inputs/AmountField';
+import Description from '../description';
+import CheckboxLabel from '../inputs/CheckboxLabel';
+import {
+  BottomCreditRetireFields,
+  RetireFormValues,
+  validateCreditRetire,
+} from './CreditRetireForm';
+import Submit from './Submit';
+import {
+  requiredMessage,
+  invalidAmount,
+  insufficientCredits,
+} from '../inputs/validation';
+import { getISOString } from '../../utils/locationStandard';
+
+/**
+ * Take - takes credits from a basket starting from the oldest credits first.
+ * https://docs.regen.network/commands/regen_tx_ecocredit_take-from-basket.html
+ *
+ * Validation:
+ *    holder: must ba a valid address, and their signature must be present in the transaction
+ *    amount: must not be empty
+ *    basket_denom: must be a valid batch denomination
+ *  if retire_on_take is true:
+ *    retirement_location: must be a valid location
+ *
+ * Also:
+ * https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.EventTake
+ */
+
+const useStyles = makeStyles((theme: Theme) => ({
+  checkboxLabel: {
+    marginTop: theme.spacing(10.75),
+    alignItems: 'initial',
+    '& .MuiCheckbox-root': {
+      alignSelf: 'end',
+    },
+  },
+  checkboxDescription: {
+    color: theme.palette.primary.contrastText,
+    fontSize: theme.spacing(4.5),
+    fontWeight: 'bold',
+    marginLeft: theme.spacing(1),
+  },
+}));
+
+// Output (submit)
+export interface MsgTakeValues {
+  owner: string;
+  basketDenom: string;
+  amount: string;
+  retireOnTake: boolean;
+  retirementLocation?: string;
+  retirementNote?: string;
+}
+
+interface CreditTakeFormValues {
+  amount: number;
+  retireOnTake?: boolean;
+  note?: string;
+  country: string;
+  stateProvince: string;
+  postalCode?: string;
+  retirementLocation?: string;
+}
+
+// Input (args)
+interface FormProps {
+  mapboxToken: string;
+  accountAddress: string;
+  basket: Basket;
+  availableTradableAmount: number;
+  onClose: () => void;
+  onSubmit: (values: MsgTakeValues) => void;
+}
+
+const BasketTakeForm: React.FC<FormProps> = ({
+  mapboxToken,
+  accountAddress,
+  basket,
+  availableTradableAmount,
+  onClose,
+  onSubmit,
+}) => {
+  const styles = useStyles();
+
+  const initialValues = {
+    amount: 0,
+    retireOnTake: !basket.disableAutoRetire,
+    note: '',
+    country: 'US',
+    stateProvince: '',
+    retirementLocation: undefined,
+  };
+
+  const validateHandler = (
+    values: CreditTakeFormValues,
+  ): FormikErrors<CreditTakeFormValues> => {
+    let errors: FormikErrors<CreditTakeFormValues> = {};
+
+    if (!values.amount) {
+      errors.amount = requiredMessage;
+    } else if (Math.sign(values.amount) !== 1) {
+      errors.amount = invalidAmount;
+    } else if (values.amount > availableTradableAmount) {
+      errors.amount = insufficientCredits;
+    }
```

Comment:
you could use `validateAmount` instead, see in `BasketPutForm`

Reply:
done

Reply:
added custome message for basket tokens case